cats_frontend/src/app/modules/cats/pages/cats-list/cats-list.component.ts

```diff
@@ -59,144 +61,117 @@ export class CatsListComponent implements OnInit {
         description: new FormControl('', [Validators.required])
     });
 
+   
+
     constructor(
-        private catsService: CatsService,
-        private commentsService: CommentsService,
-        private messageService: MessageService,
         private dialogService: DialogService,
         private router: Router,
         private route: ActivatedRoute,
-        private datePipe: DatePipe
+        private datePipe: DatePipe,
+        private store: Store,
         ) { }
 
     ngOnInit(): void {
-        this.catsService.get().subscribe((data: { cats: ICat[], count: number}) => {
-            this.cats = data.cats;
-            this.count = data.count;
-        });
 
         // search from name input
         // debounce 300ms before sending the request
-        this.searchSubscription = 
-            this.searchText.valueChanges.pipe(
-                debounceTime(300),
-                distinctUntilChanged()
-            ).subscribe((_) => {
-                this.searchSubscription?.add(this.catsService.get(
-                    this.first / this.rows + 1,
-                    this.searchText.value,
-                    this.sortField
-                ).subscribe((data: { cats: ICat[], count: number}) => {
-                    this.cats = data.cats;
-                    this.count = data.count;
-                    this.first = 0;
-                }));
-            })
+        this.searchText.valueChanges.pipe(
+            takeUntil(this._unsubscribe$$),
+            debounceTime(300),
+            distinctUntilChanged()
+        ).subscribe((_) => {
+            this.store.dispatch(CatsActions.load({ params: {
+                page: this.first / this.rows + 1,
+                field: this.sortOrder == -1 ? '-' + this.sortField : this.sortField,
+                search: this.searchText.value != '' ? this.searchText.value : null,                    
+                }})
+            )
+        })
+    }
+
+    ngOnDestroy(): void {
+        this._unsubscribe$$.next();
+        this._unsubscribe$$.complete();
     }
 
     // lazy load call
     // each filters and sorting event 
     // will trigger this function
-    loadCat($event: any) {
-
-        const filters = $event.filters;
+    loadCat($event: any): void {
 
+        let filters = cloneDeep($event.filters);
         this.first = $event.first;
         this.sortField = $event.sortField;
         this.sortOrder = $event.sortOrder;
 
-        this.catsService.get(
```

Comment:
same here only dispatch an new action of loading when something on the table change